lighthouse-core/audits/work-during-interaction.js

```diff
@@ -0,0 +1,252 @@
+
+        results.push({
+          url: url,
+          total: totalExecutionTimeForURL,
+          scripting: scriptingTotal,
+          layout: layoutTotal,
+          render: renderTotal,
+        });
+      }
+
+      const filteredResults = results
+        .filter(result => result.total > TASK_THRESHOLD)
+        .sort((a, b) => b.total - a.total);
+
+      items.push({
+        phase: str_(UIStrings[/** @type {keyof UIStrings} */ (phaseName)]),
+        total: (phaseTimes.endTs - phaseTimes.startTs) / 1000,
+        subItems: {
+          type: 'subitems',
+          items: filteredResults,
+        },
+      });
+    }
+
+    return {
+      items,
+      phases,
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const {settings} = context;
+    // TODO: responsiveness isn't yet supported by lantern.
+    if (settings.throttlingMethod === 'simulate') {
+      return {score: null, notApplicable: true};
+    }
+
+    const trace = artifacts.traces[WorkDuringInteraction.DEFAULT_PASS];
+    const metricData = {trace, settings};
+    const interactionEvent = await ComputedResponsivenes.request(metricData, context);
+    // If no interaction, diagnostic audit is n/a.
+    if (interactionEvent === null) {
+      return {score: null, notApplicable: true};
+    }
+
+    // Network records will usually be empty for timespans.
+    const devtoolsLog = artifacts.devtoolsLogs[WorkDuringInteraction.DEFAULT_PASS];
+    const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+
+    const processedTrace = await ProcessedTrace.request(trace, context);
+    const {phases, items} = await WorkDuringInteraction.mainThreadBreakdown(
+          interactionEvent, trace, processedTrace, networkRecords);
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [
+      /* eslint-disable max-len */
+      {key: 'phase', itemType: 'text', subItemsHeading: {key: 'url', itemType: 'url'}, text: 'Phase'},
+      {key: 'total', itemType: 'ms', subItemsHeading: {key: 'total', granularity: 1, itemType: 'ms'}, text: 'Total time'},
+      {key: null, itemType: 'ms', subItemsHeading: {key: 'scripting', granularity: 1, itemType: 'ms'}, text: 'Script evaluation'},
+      {key: null, itemType: 'ms', subItemsHeading: {key: 'layout', granularity: 1, itemType: 'ms'}, text: taskGroups.styleLayout.label},
+      {key: null, itemType: 'ms', subItemsHeading: {key: 'render', granularity: 1, itemType: 'ms'}, text: taskGroups.paintCompositeRender.label},
+      /* eslint-enable max-len */
+    ];
+
+    const duration = interactionEvent.args.data.duration;
+    const interactionType = interactionEvent.args.data.type
+    const displayValue = str_(UIStrings.displayValue, {timeInMs: duration, interactionType});
+
+    return {
+      score: duration < MAX_DURATION_THRESHOLD ? 1 : 0,
```

Comment:
This pass/fail condition is directly tied to the actual INP value. It seems like this audit is similar to `layout-shift-elements` because it provides a detailed breakdown of the metric value. WDYT about making this an informative audit just like `layout-shift-elements`?

Reply:
I forget the argument now for `layout-shift-elements`, but it would seem strange to me not to put a pass/fail on this audit. If it's a slow interaction we should highlight it in red as an important diagnostic to focus on.

However it doesn't make sense to use an arbitrary threshold, I had it in there for testing and forgot about it. I updated to pull the p10 from the INP audit.